src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceContext.java

```diff
@@ -68,6 +73,12 @@
     @NonNull
     private Set<ChangeRequestCheckoutStrategy> forkPRStrategies = EnumSet.noneOf(ChangeRequestCheckoutStrategy.class);
     /**
+     * Strategies used to notify Github of build status.
+     *
+     * @since TODO
+     */
+    private List<AbstractGitHubNotificationStrategy> notificationStrategies = new ArrayList<>();
```

Comment:
I think we want the default list to be `new DefaultGitHubNotificationStrategy()`

Reply:
I also think we want this list to be `final` to save GC pressure.

Reply:
Trying multiple notification strategies at once and it's tricky to 'opt out' of sending the default notification this way. The current only available method is to replace the strategies list using withNotificationStrategies(), but this would remove any strategy that happened to be loaded before.